modules/lib/core/index.d.ts

```diff
@@ -355,46 +356,47 @@ export type HttpFilterNext<
 > = (request: RequestToJava) => ResponseToJava;
 
 export interface WebSocketSession {
-    id: string
-    path: string
-
-    // TODO Maybe Record<string, string | string[]>
-    // See com.enonic.xp.portal.impl.mapper.WebSocketEventMapper
-    // And com.enonic.xp.portal.impl.mapper.MapperHelper
-    params: Record<string, unknown>
-
-    user: Omit<User,'type'>
+    id: string;
+    params: Record<string, string | string[]>;
+    path: string;
+    user: Omit<User,'type'>;
 }
 
 export type WebSocketEventType = 'open' | 'message' | 'error' | 'close';
 
 export interface WebSocketEvent<T> {
```

Comment:
My first implementation was similar to that, because it was based on the data I was able to generate in real life. I just simplified after looking at the Java class behind it.